App.js

```diff
@@ -1,123 +1,338 @@
+    constructor() {
+        super();
+        this.state = {
+            userId: "...",
+            sdkVersion: "...",
+            userActivity: "...",
+            data: []
+        };
+
+        this.setupSdk();
+    }
+
+    componentDidMount() {
+        this.subscribe();
+        this.requestPermissionAndroid();
+    }
+
+    componentWillUnmount() {
+        this.unSubscribe();
+    }
+
+    unSubscribe() {
+        this.sdkStatusSubscription.remove();
+        this.sdkUserActivityUpdateSubscribtion.remove();
+    }
+
+    subscribe() {
+        this.sdkStatusSubscription = rnSentianceEmitter.addListener(
+            "SDKStatusUpdate",
+            sdkStatus => this.onSdkStatusUpdate(sdkStatus)
+        );
+
+        this.sdkUserActivityUpdateSubscribtion = rnSentianceEmitter.addListener(
+            "SDKUserActivityUpdate",
+            userActivity => {
+                this.onUserActivityUpdate(userActivity);
+            }
+        );
+    }
+
+   async setupSdk() {
+        var shouldInitialize = await this.shouldInitialize();
+        if (shouldInitialize) {
+            await RNSentiance.init(
+                'APP_ID',
+                "APP_SECRET",
+                null,
+                true,
+            );
+        }
+
+        var sdkInitialized = await this.isSdkInitiazed();
+        if (sdkInitialized) {
+            const sdkStatus = await RNSentiance.getSdkStatus();
+            this.onSdkStatusUpdate(sdkStatus);
+            this.updateUserIdAndSdkVersion();
+            this.listenToUserActivity();
+        }
+    }
+
+    async onSdkStatusUpdate(sdkStatus) {
+      console.log("onSdkStatusUpdate");
+      const data = await this.statusToData(sdkStatus);
+      this.setState({
+        data: data
+      });
+
+      if (this.isSdkInitiazed()) {
+        this.updateUserIdAndSdkVersion();
+        this.listenToUserActivity();
+      }
+    }
+
+    onUserActivityUpdate(userActivity) {
+        var {
+            type,
+            tripInfo,
+            stationaryInfo
+        } = userActivity;
+
+        if (type === "USER_ACTIVITY_TYPE_STATIONARY") {
+            var {
+                location
+            } = stationaryInfo;
+            if (location) {
+                var {
+                    latitude,
+                    longitude
+                } = location;
+                stationaryLocation = parseFloat(latitude) + "," + parseFloat(longitude);
+            }
+
+            info = "Stationary  @" + stationaryLocation;
+            this.setState({
+                userActivity: info
+            });
+        } else if (type === "USER_ACTIVITY_TYPE_TRIP") {
+            this.setState({
+                userActivity: "Trip"
+            });
+        } else if (type === "USER_ACTIVITY_TYPE_UNKNOWN") {
+            this.setState({
+                userActivity: "Unknown"
+            });
+        }
+    }
+
+    async updateUserIdAndSdkVersion() {
+        await RNSentiance.getUserId()
```

Comment:
You can just write

`
try {
  const userId = await RNSentiance.getUserId();
} catch (err) {
  ...
}
`